scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -309,6 +310,53 @@ object DynamicTLSPeerGroup {
     */
   case class IncomingConnectionThrottlingConfig(throttleLocalhost: Boolean, throttlingDuration: FiniteDuration)
 
+  /**
+    * Configures detection on idle peer
+    * @param timeUnit unit of time for all time configs
+    * @param readerIdleTime a ChannelIdle event whose state is ReaderIdle will be triggered when
+    *                       no read was performed for the specified period of time on given channel. Specify 0 to disable.
+    * @param writerIdleTime a ChannelIdle event whose state is WriterIdle will be triggered when
+    *                       no write was performed for the specified period of time on given channel. Specify 0 to disable.
+    * @param allIdleTime a ChannelIdle event whose state is AllIdle will be triggered when
+    *                       no read was performed for the specified period of time on given channel. Specify 0 to disable.
```

Comment:
I didn't expect this: when I saw `AllIdle` I thought it means both read and write have gone idle, which would mean the timeout for it is `max(readerIdleTime, writerIdleTime)`, and if there's a write or a read than it goes back to just the other one being idle. 

BTW it says "no read was performed".

Reply:
ahh it is an typo, it should read `neither read nor write was performed ` , so it is separate idle event idicating that nothing happened on this channel over this amount of time